knowledge-portal/central/initiate_push.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# changed files can be obtained by git diff between latest commit and unstaged area
+changed_files=$(git diff --name-only HEAD)
+echo "Files changed in knowledge portal: ${changed_files}";
+
+if [[ $changed_files ]]
+then
+    # Reverse the git reset and move files to staged as we will need latest files 
+    git add -A && git commit -m "docs:reversing soft reset as now we have the filenames"
+
+    # Create an array of all changed files and pass to python script as argument
+    declare -a changed_files_array
+
+    # Run python script to get output in format "source_repo_name:filepath_central_gitrepo-->filepath_source_gitrepo"
+    while read
+    do  
+        echo "$REPLY"
+        changed_files_array+=( "$REPLY" )
+    done <<< ${changed_files}
+    echo "${changed_files_array}"
+    output=$(python3 -c "$(curl -fsSL https://github.com/avinashtrivedi11/Gitlab-CI-CD-Templates/raw/main/knowledge-portal/central/reverse_mapping.py)" ${changed_files_array})
+    echo "${output}"
+    # Create an array to keep track of all unique source repos in which files have been modified
+    declare -a source_repo_array
+
+    # From python output, separate line to corresponding variables 
+    while read line 
+    do
+        split_output=(${line//":"/ })
+        source_repo_name=${split_output[0]}
+        rhs_path="${line%-->*}"
+        lhs_path="${line##*-->}"
+        echo "lhs_path:${lhs_path}, rhs_path:${rhs_path}, source_repo_name:${source_repo_name}"
+
+        # Create a unique directory for each source repo and copy files
+        unique_directory="/root/source/${source_repo_name}"
+        
+        # Get source_repo_path by changing '_' to '/'
+        source_repo_path=${source_repo_name//'_'/'/'} 
+        
+        if [ -d "$unique_directory" ]
+        then
+            # Remove existing and Copy changed files to unique_directory
+            rm -rfv ${unique_directory}/${lhs_path}
+            cp -R -v ./"${rhs_path}" "${unique_directory}/${lhs_path}"
+        else
+            # Create directory, clone source repo, create branch, and initiate copy
+            mkdir -p ${unique_directory}
+            source_repo_array+=( "${source_repo_name}" )
+            git clone -v https://oauth2:${BOT_ACCESS_TOKEN}@gitlab.com/DetectTechnologies/${source_repo_path}.git ${unique_directory}
```

Comment:
Can we document in the readme that this `BOT_ACCESS_TOKEN` should also be defined as a variable by the maintainers centrally in the org-wide gitlab instance settings?

Reply:
Done